### DISCUSSION TITLE
@types for protobuf-fieldmask

### DISCUSSION THREAD

**Discussion by jyeros** [DefinitelyTyped/DefinitelyTyped#66198] [Request a new @types package]
Add @types for [protobuf-fieldmask](https://www.npmjs.com/package/protobuf-fieldmask).

It should be simple, maybe something like:
```ts
declare module 'protobuf-fieldmask' {
	declare type WithFieldMask<T> = { [K in keyof T]?: WithFieldMask<T[K]> | undefined };
	function applyFieldMask<T>(sourceObject: T, fieldMask: string[]): WithFieldMask<T>;
	function generateFieldMask(object: unknown): string[];
}

```

**Comment by jbovone** [comment]
@jyeros hi. 

I don't think there is a method that allows you to decompose a ["f.a"] string spec, to turn an object like:  {f: {a: 3}, [K: any]:any} into {f: {a: 3}}. 

The original code includes javascript docs. And the proper type of that function is like the js docs says. It depends on the caller to cast object to the proper type.
